test/OffsetHelper.test.ts

```diff
@@ -121,7 +125,102 @@ describe("OffsetHelper", function () {
     });
   });
 
-  describe("Testing autoOffset()", function () {
+  const TOKEN_POOLS = [
+    { name: "NCT", token: () => nct },
+    { name: "BCT", token: () => bct },
+  ];
+
+  describe("#autoOffsetUsingToken(address,uint256,address)[fixed-in]", function () {
+    async function retireFixedInToken(
+      fromToken: IERC20,
+      fromAmount: BigNumber,
+      toToken: IToucanPoolToken
+    ) {
+      const expOffset = await offsetHelper[
+        "calculateExpectedPoolTokenAmount(address,uint256,address)"
+      ](fromToken.address, fromAmount, toToken.address);
+      // sanity check, should easily be > 1 tonne for all provided inputs
+      expect(expOffset).to.be.greaterThan(ONE_ETHER);
+
+      await fromToken.approve(offsetHelper.address, fromAmount);
+
+      const supplyBefore = await toToken.totalSupply();
+      const autoOffsetUsingToken =
+        offsetHelper["autoOffsetUsingToken(address,uint256,address)"];
+      await expect(
+        autoOffsetUsingToken(fromToken.address, fromAmount, toToken.address)
+      )
+        .to.emit(offsetHelper, "Redeemed")
+        .withArgs(
+          addr2.address,
+          toToken.address,
+          anyValue,
+          (amounts: BigNumber[]) => {
+            return expOffset == sumBN(amounts);
+          }
+        )
+        .and.to.changeTokenBalance(
+          fromToken,
+          addr2.address,
+          fromAmount.mul(-1)
+        );
+
+      const supplyAfter = await toToken.totalSupply();
+      expect(supplyBefore.sub(supplyAfter)).to.equal(expOffset);
+    }
+
+    TOKEN_POOLS.forEach((pool) => {
```

Comment:
I love this idea of looping over pools. I think it should be implemented in a follow-up PR for all tests where it makes sense.

Reply:
I also think that the other tests could benefit from being moved to this format, I try to avoid repetition wherever possible :) But I didn't want to do that in this PR, it's a nice follow-up for someone else, together with using `loadFixture` (see other comment).

Reply:
> it's a nice follow-up for someone else

I agree, I'll open an issue once we merge this so I can link to the commit the introduce this